sim/src/ics/compVec.test.cpp

```diff
@@ -0,0 +1,68 @@
+#include <gtest/gtest.h>
+#include "component.h"
+#include "compVec.h"
+
+#define TEST_SUITE CompVecTest
+
+using namespace ics;
+
+struct ParentComp : public DefaultComponent {
+    int height;
+};
+
+struct ChildComp : public ParentComp {
+    int width;
+};
```

Comment:
Are we going to support component inheritance on the user API?

Reply:
No, I made this because I was worried about casting issues which I hope to test later on. I had some bad experiences with the previous engine when this happens:
```
compVec<ChildComp> x = ...;
// Somehow needs to be converted to compVec<ParentComp>
compVec<ParentComp> y = x;
// Vectors will have problems accessing the correct memory locations because ChildComp is larger than ParentComp
y.at(3)
```

Reply:
Parent and child component implies some of superior relationship. Also its not immediately clear how parent component is used us there is no test that uses it. Perhaps a rename and some comments to describe what u are trying todo?

Reply:
I think I will just remove one of them and rename accordingly. I can test the issue I described above, later.

Reply:
Fixed in 59a3e0a